### DISCUSSION TITLE
[firebase_storage] Object does not exist at location

### DISCUSSION THREAD

**Discussion by renefloor** [firebase/flutterfire#6899] [Q&A]
I don’t use really use firebase_storage myself, but still support a cache manager using it 😅.
Someone (@Nolence) has this issue: https://github.com/Baseflow/flutter_cache_manager/issues/240
He gets an `Unhandled Exception: PlatformException(download_error, Object does not exist at location., null, null)` for a valid gs:// url. 
I’m not sure what is going wrong here and if his proposed solution is really the right way to go. I think it just shouldn’t go wrong this way and it might be a one off.
He replaces this:
```
    var ref = FirebaseStorage.instance.ref().child(url);
    var _url = await ref.getDownloadURL() as String;
```
with this
```
    final ref = await firebaseStorage.getReferenceFromUrl(audioURL);
```

His PR contains a different proposal though to also get more meta data.

I want to keep the library as simple as possible and this solution makes it more complex, so I prefer not to do it, but also don’t want him to have a bug.

Does anybody know why the `Object does not exist at location` error could occur by calling `getDownloadURL()`?

**Comment by mikehardy** [comment]
Possibly related #6848

released in  firebase_storage_web-v3.0.2 firebase_storage_platform_interface-v4.0.2 firebase_storage-v10.0.3  https://github.com/FirebaseExtended/flutterfire/commit/d7e81b3aa0fe4ebd1a84fa10ebe5026e47ef3791

I wonder what version the affected person is on, and if they could share a specific URL that does not work (along with the bucket gs:// it should map to perhaps?)

**Reply by stargazing-dino** [reply]
ah @mikehardy , sorry, wrong reply button. Meant to reply to chat. Yes, I'll get specific urls, versions for you this weekend if possible

**Reply by mikehardy** [reply]
Thanks! The URL->GS bucket mapping in particular is vital to move this forward - we have some known-should-work mappings that we use in automated tests and as soon as we have a failure mapping we can get right to a fix. Cheers

**Reply by mikehardy** [reply]
Just a gentle reminder @Nolence, this is an area of current work for us in a couple repos so we are very curious for any test URLs/GS mappings that are problematic. We'd love to get them right. Thanks!

**Comment by stargazing-dino** [comment]
Heyo, ya, I was hesitant about adding in all that metadata stuff myself. I do prefer simplicity but it seemed necessary for that method. I know the PR has fixed my immediate issue but the fact not that many other people have echoed my error it might be a version thing. I'll do a proper test on a fresh Flutter project later and see if it encounter anything.

Yes though, I'd love to hear how the firebase team officially recommends to download resources as well. Was very curious when I looked into that pr and never found anything solid.
